shoes-core/spec/shoes/builtin_methods_spec.rb

```diff
@@ -22,63 +22,35 @@
     end
   end
 
-  describe "info" do
-    before :each do
-      allow(logger).to receive(:info)
-      app.info("test")
-    end
-
-    it "sets Shoes::LOG" do
-      expect(Shoes::LOG).to eq([["info", "test"]])
-    end
-
-    it "sends message to logger" do
-      expect(logger).to have_received(:info)
-    end
-  end
-
-  describe "debug" do
-    before :each do
-      allow(logger).to receive(:debug)
-      app.debug("test")
-    end
-
-    it "sets Shoes::LOG" do
-      expect(Shoes::LOG).to eq([["debug", "test"]])
-    end
-
-    it "sends message to logger" do
-      expect(logger).to have_received(:debug)
-    end
-  end
-
-  describe "warn" do
-    before :each do
-      allow(logger).to receive(:warn)
-      app.warn("test")
-    end
+  %w(info debug warn error).each do |level|
+    describe "##{level}" do
+      before :each do
+        allow(logger).to receive(level)
+        app.send(level, "test")
```

Comment:
I'd prefer `public_send` here, otherwise we might make them private and not even notice :)